src/components/pages/ReadArticle.js

```diff
@@ -14,27 +23,79 @@ class ReadArticle extends Component {
   state = {
     Article: {},
     Author: {},
+    AllBoomarked: {},
+    userId: {},
+    isBookmarked: false,
+    isProfileRequested: false,
+    isPreviousBookmarked: false,
+    redirect: false,
   };
 
   componentWillMount() {
     const { slug } = this.props.match.params;
     this.props.getArticle(slug);
+    this.props.getBoomarks(slug);
   }
 
   componentWillReceiveProps(newProps) {
     if (newProps.Article) {
       this.setState({ Article: newProps.Article });
-      this.props.getUserProfile(newProps.Article.article.author.username);
+      if (!this.state.isProfileRequested) {
+        this.props.getUserProfile(newProps.Article.article.author.username);
+        this.setState({ isProfileRequested: true });
+      }
     }
     if (newProps.Author) {
       this.setState({ Author: newProps.Author });
     }
+    if (newProps.myBookmarks.length > 0) {
+      const { slug } = this.props.match.params;
+      if (
+        this.isThisSlugBookmarked(slug, newProps.myBookmarks)
+        && !this.state.isPreviousBookmarked
+      ) {
+        this.setState({ isBookmarked: true, isPreviousBookmarked: true });
+      }
+    }
   }
 
+  handleClick = () => {
```

Comment:
@EmyRukundo make function names more descriptive and readable enough. handleClick, yes, but which click and on which component/button?

Reply:
Thanks for the feedback, I am going to correct it